src/extension.ts

```diff
@@ -0,0 +1,75 @@
+/**
+ * d2 File Viewer Extension
+ **/
+
+import {
+	workspace,
+	TextDocumentChangeEvent,
+	TextDocument,
+	ExtensionContext,
+	commands,
+	window,
+	WorkspaceConfiguration
+} from 'vscode';
+
+import { DocToPreviewGenerator } from './docToPreviewGenerator';
+
+const d2Ext = 'd2';
+
+let previewGenerator: DocToPreviewGenerator = new DocToPreviewGenerator();
+let ws: WorkspaceConfiguration = workspace.getConfiguration('d2-viewer');
+
+export function activate(context: ExtensionContext): void {
+
+	context.subscriptions.push(workspace.onDidChangeTextDocument((e: TextDocumentChangeEvent) => {
+		if (e.document.languageId === d2Ext) {
+			let autoUp = ws.get('autoUpdate', false);
+
+			if (autoUp) {
+				let trk = previewGenerator.getTrackObject(e.document);
+				trk?.timer?.reset();
```

Comment:
do we need the refreshTimer, can't we just run each time this TextDocumentChangeEvent is fired (possibly with a debounce if it is too frequent)?

Reply:
The timer is the debounce, if a change is made, then another one isn't detected for the given time (1.5 sec by default), then update occurs.  The reset function just says, "I know your counting, but start over, there are more changes coming"

Reply:
hmm okay that makes sense I think we can use `setTimeout` for the debounce instead of `setInterval` and then we shouldn't need to call `this.stop()` and can just call `this.callback()` since it would only run once by default. It makes it clearer to me that it isn't going to be run at a regular interval which is how I misread it initially.